### PR TITLE
Add __repr__ and __str__ to core.py classes

### PR DESCRIPTION
**Pull request by OwenGrannis**
# Pull Request
### Description
Added __repr__ and __str__ methods to all non-abstract classes in core.py

This pull request partially addresses #13 

- [ ] make sure you are pulling from a dev branch on your fork
- [ ] check for code linting

### CODE REVIEW

**Comment by arichar6** [comment]
First, thanks for taking this on. I think that having readable __repr__ and __str__ outputs should help make turboPy more user friendly.

I have a few thoughts, though. Here they are in no particular order:

- I would like `__repr__` to return a string that you could (at least theoretically) copy and paste into a python console to get an instance that is (as close as reasonable) to the given object. Since these classes have some complicated internal states, at least for now I don't think we should be concerned about that. Just give something that would make the object like it was when initialized. E.g., for the `Simulation` class you could have:
```python
    def __repr__(self):
        return (f"{self.__class__.__name__}"
                     f"({self.input_data})")
```
Since most of these classes just take one input parameter (which is a dictionary that is saved), the `__repr__` function should be pretty straightforward.
- I'd like to start by just implementing `__repr__`, unless there is some very good reason to also implement `__str__`. Note that `__str__` will just call `__repr__`, which is a nice default. One reason that I say this is that, while the extra description of the object is nice, it ends up duplicating information that should be in the docstring. So, if a user needs more info about the Grid class, they can just type `help(turbopy.Grid)`, and get the info from the docstring.
- I have a minor change to the `PhysicsModule.__repr__` method. See comment in review.

File: turbopy/core.py [hunk 1]
```diff
@@ -419,6 +433,14 @@ def initialize(self):
         """
         pass
 
+    def __repr__(self):
+        representation = f"{self.__class__.__name__}{self.module_type}, {self.input_data})"
+        return representation
+
+    def __str__(self):
+        base_str = f"A PhysicsModule with name {self.module_type}."
+        return base_str
```
Comment (arichar6) [inline turbopy/core.py:442]: There is a parenthesis missing in `__repr__`. Also, `module_type` is not necessarily defined (i.e., could be `None`).
```suggestion
    def __repr__(self):
        return f"{self.__class__.__name__}({self.input_data})"
```

I don't think we need `__str__` right now.

File: turbopy/core.py [hunk 2]
```diff
@@ -798,3 +818,6 @@ def finalize(self):
         complete.
         """
         pass
+    
+    def __repr__(self):
+        representation = f"{self.__class__.__name__}({self.input_data})"
```
Comment (arichar6) [inline turbopy/core.py:823]: This needs to return something.

File: turbopy/core.py [hunk 3]
```diff
@@ -540,6 +550,14 @@ def is_running(self):
         """Check if time is less than end time"""
         return self.this_step < self.num_steps
 
+    def __repr__(self):
+        clock_data = {
+            "start_time" : self.start_time,
+            "end_time" : self.end_time,
+            "num_steps" : self.num_steps,
+            "print_time" : self.print_time}
+        representation = f"{self.__class__.__name__}({clock_data})"
```
Comment (arichar6) [inline turbopy/core.py:559]: You should just change `__init__` to save `clock_data` as `self.clock_data`. Then you can use `self.clock_data` here. This way, if future changes to this class introduce new parameters, this `__repr__` will still work.

File: turbopy/core.py [hunk 4]
```diff
@@ -821,3 +817,4 @@ def finalize(self):
     
     def __repr__(self):
         representation = f"{self.__class__.__name__}({self.input_data})"
```
Comment (arichar6) [inline turbopy/core.py:819]:
```suggestion
        return f"{self.__class__.__name__}({self.input_data})"
```

File: turbopy/core.py [hunk 5]
```diff
@@ -551,12 +552,7 @@ def is_running(self):
         return self.this_step < self.num_steps
 
     def __repr__(self):
-        clock_data = {
-            "start_time" : self.start_time,
-            "end_time" : self.end_time,
-            "num_steps" : self.num_steps,
-            "print_time" : self.print_time}
-        representation = f"{self.__class__.__name__}({clock_data})"
+        representation = f"{self.__class__.__name__}({self.clock_data})"
```
Comment (arichar6) [inline turbopy/core.py:555]: Nice!

**Comment by arichar6** [comment]
Also, there have been changes to `main` since you started this. Some changes are causing merge conflicts. Please resolve those, too.

**Comment by OwenGrannis** [comment]
> Also, there have been changes to `main` since you started this. Some changes are causing merge conflicts. Please resolve those, too.

How exactly should I fix it? Should I pull request to my personal repo to update it, then pull that down to my personal git?

**Comment by arichar6** [comment]
I took care of the merge conflicts. I just used the GitHub web editor, which is probably the fastest way.

**Pull request merged**